### PR TITLE
Replace the GraphQL-to-Typescript codegen library

### PR DESCRIPTION
**Pull request by bystam**
# Jira Issue: [SC2-19]

## What?
- Change the graphql-to-typescript codegen library to a more common and maintained one with way more features
- https://www.graphql-code-generator.com/docs/plugins/typescript

## Why?
- The one we were using had way too few features
- Better outlined the ticket: https://hedvig.atlassian.net/secure/RapidBoard.jspa?rapidView=17&projectKey=SC2&selectedIssue=SC2-19

## Optional checklist
- [ ] Unit tests written
- [x] Tested locally
- [x] Codescouted

### CODE REVIEW

File: codegen.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+schema: "./src/schema.graphqls"
+documents: null
+generates:
+  src/generated/graphql.ts:
+    plugins:
+      - "typescript"
+      - "typescript-resolvers"
+    config:
+      contextType: ../context#Context
+      noSchemaStitching: true
```
Comment (bystam) [inline codegen.yml:11]: This has to do with the codegen itself, not our stitching outside of it.

Adding this created a LOT of more union types for all resolvers, which we didn't need and would have had to work around

File: codegen.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      maybeValue: T | undefined
```
Comment (bystam) [inline codegen.yml:12]: It's `T | null` by default, but that was incompatible with a bunch of our existing code so I changed it to this

File: Dockerfile [hunk 3]
```diff
@@ -17,7 +17,7 @@ FROM build AS test
 
 RUN yarn test
 
-FROM node:10.14.2-alpine AS assemble
+FROM node:16.4.0-alpine AS assemble
```
Comment (bystam) [inline Dockerfile:20]: We were using veeeery old Node versions :) 

https://nodejs.org/en/about/releases/

File: src/resolvers/currentChatResponse.ts [hunk 4]
```diff
@@ -2,29 +2,31 @@ import { equals } from 'ramda'
...
   const token = getToken()
   const chat = await getChat(token, headers)
-  return transformMessage(chat.messages[0])
+
+  // fields identical for Message and ChatResponse, only need to set typename
+  return {...transformMessage(chat.messages[0]), __typename: "ChatResponse"}
 }
 
-export const subscribeToCurrentChatResponse: SubscriptionToCurrentChatResponseResolver = {
+export const subscribeToCurrentChatResponse: SubscriptionResolvers['currentChatResponse'] = {
```
Comment (vonElfvin) [inline src/resolvers/currentChatResponse.ts:24]:
```suggestion
export const currentChatResponseSubscription: SubscriptionResolvers['currentChatResponse'] = {
```

File: src/resolvers/index.ts [hunk 5]
```diff
@@ -1,146 +1,130 @@
...
-import { chatActions } from './chatActions'
+  bankIdAuth,
+  danishBankIdAuth,
+  norwegianBankIdAuth,
+  subscribeToAuthStatus,
+  swedishBankIdAuth
+} from '../features/auth'
+import {createOffer, getInsuranceByOfferSuccessEvent, signOffer, subscribeToOffer,} from '../features/offer'
+import {getSignStatus, getSignStatusFromSignEvent, signOfferV2, subscribeToSignStatus,} from '../features/offer/sign'
+import {registerBranchCampaign} from './analytics'
```
Comment (vonElfvin) [inline src/resolvers/index.ts:10]: `prettier` 😢 
```suggestion
import { registerBranchCampaign } from './analytics'
```

File: src/resolvers/selectCashbackOption.ts [hunk 6]
```diff
@@ -18,7 +19,7 @@ const selectCashbackOption: MutationToSelectCashbackOptionResolver = async (
 
   if (result === 204 || result === 200) {
     const cashback = await cashbackInner(token, headers)
-    if (cashback !== null) {
+    if (cashback) {
```
Comment (vonElfvin) [inline src/resolvers/selectCashbackOption.ts:22]: Nice catch

**Review by vonElfvin — APPROVED** [review]
Neat! This looks like a big improvement :)

File: src/features/auth/auth.ts [hunk 7]
```diff
@@ -1,10 +1,10 @@
-import { swedishAuthMember, norwegianAuthMember, danishAuthMember } from '../../api'
-import { MutationToBankIdAuthResolver, MutationToSwedishBankIdAuthResolver, MutationToNorwegianBankIdAuthResolver, MutationToDanishBankIdAuthResolver } from '../../typings/generated-graphql-types'
+import {danishAuthMember, norwegianAuthMember, swedishAuthMember} from '../../api'
```
[the remark this answers is not in the document]
Comment (bystam) [reply src/features/auth/auth.ts:1]: Yeah I think this comes from Antons auto-formatting. What did you use to do it, @anthansson?

File: src/resolvers/addressChange.test.ts [hunk 8]
```diff
@@ -85,7 +80,7 @@ describe('createAddressChangeQuotes - Denmark', () => {
 // Helpers
 
 const MUTATIONS = {
-  createAddressChangeQuotes: async (input: AddressChangeInput) =>
+  createAddressChangeQuotes: async (input: any) =>
```
[the remark this answers is not in the document]
Comment (bystam) [reply src/resolvers/addressChange.test.ts:83]: Designed to not import any of the resolver types, similarly to how I like to write the tests for spring backends.

**Pull request merged**